api/pkg/workflow/edge_test.go

```diff
@@ -0,0 +1,37 @@
+package workflow
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEdge_SetID(t *testing.T) {
+	e := &Edge{}
+	e.SetID("id")
+	assert.Equal(t, "id", e.id)
+}
+
+func TestEdge_SetFrom(t *testing.T) {
+	e := &Edge{}
+	e.SetFrom([]string{"from"})
+	assert.Equal(t, []string{"from"}, e.from)
+}
+
+func TestEdge_SetTo(t *testing.T) {
+	e := &Edge{}
+	e.SetTo([]string{"to"})
+	assert.Equal(t, []string{"to"}, e.to)
+}
+
+func TestEdge_SetFromPort(t *testing.T) {
+	e := &Edge{}
+	e.SetFromPort("fromPort")
+	assert.Equal(t, "fromPort", e.fromPort)
+}
+
+func TestEdge_SetToPort(t *testing.T) {
+	e := &Edge{}
+	e.SetToPort("toPort")
+	assert.Equal(t, "toPort", e.toPort)
+}
```

Comment:
In general, for such simple setters, it's often sufficient to test them indirectly through other, more complex methods that use them. This approach provides better test coverage of actual behavior while reducing the overhead of maintaining many small, simple tests.

For simple setters like these, You go with Combine tests i.e Instead of individual tests for each setter, you could have a single test that checks all setters.(Just an example we don't we even need this), we might need something like this if functions get too complicated.

```suggestion
package workflow

import (
    "testing"
    "github.com/stretchr/testify/assert"
)

func TestEdgeSetters(t *testing.T) {
    e := &Edge{}
    e.SetID("testID")
    assert.Equal(t, "testID", e.id, "SetID should correctly set the id field")
    fromNodes := []string{"node1", "node2"}
    e.SetFrom(fromNodes)
    assert.Equal(t, fromNodes, e.from, "SetFrom should correctly set the from field")
    toNodes := []string{"node3", "node4"}
    e.SetTo(toNodes)
    assert.Equal(t, toNodes, e.to, "SetTo should correctly set the to field")
    e.SetFromPort("fromPort")
    assert.Equal(t, "fromPort", e.fromPort, "SetFromPort should correctly set the fromPort field")
    e.SetToPort("toPort")
    assert.Equal(t, "toPort", e.toPort, "SetToPort should correctly set the toPort field")
}
```

Reply:
Done